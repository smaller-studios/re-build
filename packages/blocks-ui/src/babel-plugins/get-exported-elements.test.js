import { testPlugin } from './test-util'
import Plugin from './get-exported-elements'

test('should update internal Plugin state with ExportNamedDeclaration definitions', () => {
  const plugin = new Plugin()
  expect(plugin.state).toEqual({ elements: {} })

  testPlugin(plugin.plugin, `export const HeaderBasic = () => <div></div>`)

  expect(plugin.state.elements).toMatchSnapshot()
})

test('should not updated internal Plugin state when no named exports in tree', () => {
  const plugin = new Plugin()
  expect(plugin.state).toEqual({ elements: {} })

  testPlugin(
    plugin.plugin,
    `const HeaderBasic = () => <div></div>
export default HeaderBasic
    `
  )

  expect(plugin.state).toEqual({ elements: {} })
})
