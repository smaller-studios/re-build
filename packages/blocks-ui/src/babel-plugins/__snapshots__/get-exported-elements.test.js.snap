// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should generate a tree of JSX elements from an AST 1`] = `
Object {
  "HeaderBasic": Node {
    "children": Array [],
    "closingElement": Node {
      "end": 44,
      "innerComments": undefined,
      "leadingComments": undefined,
      "loc": SourceLocation {
        "end": Position {
          "column": 44,
          "line": 1,
        },
        "start": Position {
          "column": 38,
          "line": 1,
        },
      },
      "name": Node {
        "end": 43,
        "innerComments": undefined,
        "leadingComments": undefined,
        "loc": SourceLocation {
          "end": Position {
            "column": 43,
            "line": 1,
          },
          "start": Position {
            "column": 40,
            "line": 1,
          },
        },
        "name": "div",
        "start": 40,
        "trailingComments": undefined,
        "type": "JSXIdentifier",
      },
      "start": 38,
      "trailingComments": undefined,
      "type": "JSXClosingElement",
    },
    "end": 44,
    "innerComments": undefined,
    "leadingComments": undefined,
    "loc": SourceLocation {
      "end": Position {
        "column": 44,
        "line": 1,
      },
      "start": Position {
        "column": 33,
        "line": 1,
      },
    },
    "openingElement": Node {
      "attributes": Array [],
      "end": 38,
      "innerComments": undefined,
      "leadingComments": undefined,
      "loc": SourceLocation {
        "end": Position {
          "column": 38,
          "line": 1,
        },
        "start": Position {
          "column": 33,
          "line": 1,
        },
      },
      "name": Node {
        "end": 37,
        "innerComments": undefined,
        "leadingComments": undefined,
        "loc": SourceLocation {
          "end": Position {
            "column": 37,
            "line": 1,
          },
          "start": Position {
            "column": 34,
            "line": 1,
          },
        },
        "name": "div",
        "start": 34,
        "trailingComments": undefined,
        "type": "JSXIdentifier",
      },
      "selfClosing": false,
      "start": 33,
      "trailingComments": undefined,
      "type": "JSXOpeningElement",
    },
    "start": 33,
    "trailingComments": undefined,
    "type": "JSXElement",
  },
}
`;
